### PR TITLE
Added book related code

### PR DESCRIPTION
**Pull request by MilianCode**

### CODE REVIEW

File: src/main/java/com/example/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -1,24 +1,44 @@
...
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto findById(Long id) {
+        Optional<Book> optionalBook = bookRepository.findById(id);
+        if (optionalBook.isPresent()) {
+            return bookMapper.toDto(optionalBook.get());
+        }
+        throw new EntityNotFoundException("Cant find book by id = " + id);
     }
```
Comment (okuzan) [inline src/main/java/com/example/service/impl/BookServiceImpl.java]:
```suggestion
     Book book = bookRepository.findById(id)
        .orElseThrow(() -> new EntityNotFoundException("Cant find book by id = " + id));
    return bookMapper.toDto(book);
```

File: src/main/java/com/example/model/Book.java [hunk 2]
```diff
@@ -4,15 +4,13 @@
...
 import jakarta.persistence.GenerationType;
 import jakarta.persistence.Id;
+import jakarta.persistence.Table;
 import java.math.BigDecimal;
-import lombok.Getter;
-import lombok.Setter;
-import lombok.ToString;
+import lombok.Data;
 
+@Data
```
Comment (okuzan) [inline src/main/java/com/example/model/Book.java]: revert those changes, @ Data contains problematic annotations for JPA entities: toString() (when related entities are lazy initalized), equalsAndHashcode(), so use only the things you need, or explicitly deal with them like using @equalsandhashcode.exclude etc

File: src/main/java/com/example/demo/DemoApplication.java [hunk 3]
```diff
@@ -2,8 +2,12 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.ComponentScan;
 
 @SpringBootApplication
+@EntityScan(basePackages = {"com.example"})
+@ComponentScan(basePackages = {"com.example"})
```
Comment (okuzan) [inline src/main/java/com/example/demo/DemoApplication.java]: why not invent something more descriptive than `com.example` 
Also I'm not sure you need them at all, if all of your classes are under the same package

File: src/main/java/com/example/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/api/books")
```
Comment (okuzan) [inline src/main/java/com/example/controller/BookController.java]: extract common to all endpoints /api prefix to application.properties (server servlet context-path)

File: src/main/java/com/example/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+public class BookController {
+
```
Comment (okuzan) [inline src/main/java/com/example/controller/BookController.java]:
```suggestion
```
don't leave empty lines after class/method definition

File: src/main/java/com/example/controller/BookController.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    private final BookService bookService;
+
+    @Autowired
+    public BookController(BookService bookService) {
+        this.bookService = bookService;
+    }
```
Comment (okuzan) [inline src/main/java/com/example/controller/BookController.java]: just use @ RequiredArgsConstructor

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/DemoApplication.java (hunk 3 above)
Comment (MilianCode) [reply src/main/java/com/example/demo/DemoApplication.java]: Without it, my application wont work properly. In console there is no hibernate requests, and any request to application has 404 not found response: {
    "timestamp": "2024-04-20T08:10:45.536+00:00",
    "status": 404,
    "error": "Not Found",
    "path": "/api/books"
}

File: src/main/java/com/miliancode/demo/DemoApplication.java [hunk 7]
```diff
@@ -1,9 +1,13 @@
...
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.ComponentScan;
 
 @SpringBootApplication
+@EntityScan(basePackages = {"com.miliancode"})
+@ComponentScan(basePackages = {"com.miliancode"})
 public class DemoApplication {
```
Comment (olekskov) [inline src/main/java/com/miliancode/demo/DemoApplication.java:11]: Move this class out of demo folder to root one and necessity of these annotation should disappear

**Review by olekskov — APPROVED** [review]

**Pull request merged**